code/modules/atmospherics/machinery/components/fusion/hfr_procs.dm

```diff
@@ -318,10 +318,106 @@
  * Create the explosion + the gas emission before deleting the machine core.
  */
 /obj/machinery/atmospherics/components/unary/hypertorus/core/proc/meltdown()
-	explosion(src, light_impact_range = power_level * 5, flash_range = power_level * 6, adminlog = TRUE, ignorecap = TRUE)
-	radiation_pulse(loc, power_level * 7000, (1 / (power_level + 5)), TRUE)
-	empulse(loc, power_level * 5, power_level * 7, TRUE)
-	var/list/around_turfs = circlerangeturfs(src, power_level * 5)
+	var/flash_explosion = 0
+	var/light_impact_explosion = 0
+	var/heavy_impact_explosion = 0
+	var/devastating_explosion = 0
+	var/em_pulse = selected_fuel.meltdown_flags & HYPERTORUS_FLAG_EMP
+	var/rad_pulse = selected_fuel.meltdown_flags & HYPERTORUS_FLAG_RADIATION_PULSE
+	var/emp_light_size = 0
+	var/emp_heavy_size = 0
+	var/rad_pulse_size = 0
+	var/rad_pulse_strength = 0
+	var/gas_spread = 0
+	var/gas_pockets = 0
+	var/critical = FALSE
+
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_BASE_EXPLOSION)
+		flash_explosion = power_level * 3
+		light_impact_explosion = power_level * 2
+
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_MEDIUM_EXPLOSION)
+		flash_explosion = power_level * 6
+		light_impact_explosion = power_level * 5
+		heavy_impact_explosion = power_level * 0.5
+
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_DEVASTATING_EXPLOSION)
+		flash_explosion = power_level * 8
+		light_impact_explosion = power_level * 7
+		heavy_impact_explosion = power_level * 2
+		devastating_explosion = power_level
+
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_MINIMUM_SPREAD)
+		if(em_pulse)
+			emp_light_size = power_level * 3
+			emp_heavy_size = power_level * 1
+		if(rad_pulse)
+			rad_pulse_size = (1 / (power_level + 1))
+			rad_pulse_strength = power_level * 3000
+		gas_pockets = 5
+		gas_spread = power_level * 2
+
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_MEDIUM_SPREAD)
+		if(em_pulse)
+			emp_light_size = power_level * 5
+			emp_heavy_size = power_level * 3
+		if(rad_pulse)
+			rad_pulse_size = (1 / (power_level + 3))
+			rad_pulse_strength = power_level * 5000
+		gas_pockets = 7
+		gas_spread = power_level * 4
+
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_BIG_SPREAD)
+		if(em_pulse)
+			emp_light_size = power_level * 7
+			emp_heavy_size = power_level * 5
+		if(rad_pulse)
+			rad_pulse_size = (1 / (power_level + 5))
+			rad_pulse_strength = power_level * 7000
+		gas_pockets = 10
+		gas_spread = power_level * 6
+
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_MASSIVE_SPREAD)
+		if(em_pulse)
+			emp_light_size = power_level * 9
+			emp_heavy_size = power_level * 7
+		if(rad_pulse)
+			rad_pulse_size = (1 / (power_level + 7))
+			rad_pulse_strength = power_level * 9000
+		gas_pockets = 15
+		gas_spread = power_level * 8
+
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_CRITICAL_MELTDOWN)
+		critical = TRUE
+		Cinematic(CINEMATIC_SELFDESTRUCT,world)
```

Comment:
I hate to say this given the context, but I think cinematics should be reserved for roundenders, or fakeout memes. Partially because they steal vision during what could be an important period, and partially because I think otherwise they kinda lose their impact

Reply:
If i understand it's for real big explosion, we can have just station wide siren or something, still notifies about such explosion, but doesn't take all screen